From: Hong Zhiguo <zhiguohong@tencent.com>
Date: Tue, 22 Oct 2013 18:32:56 +0000
Subject: e1000: fix wrong queue idx calculation
Patch-mainline: v3.13-rc1
Git-commit: 49a45a0686cc2b43bcb3834a68416a201475dc77
References: bnc#857400 FATE#315649

tx_ring and adapter->tx_ring are already of type "struct
e1000_tx_ring *"

Signed-off-by: Hong Zhiguo <zhiguohong@tencent.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/intel/e1000/e1000_main.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)
--- a/drivers/net/ethernet/intel/e1000/e1000_main.c
+++ b/drivers/net/ethernet/intel/e1000/e1000_main.c
@@ -3912,8 +3912,7 @@ static bool e1000_clean_tx_irq(struct e1
 			      "  next_to_watch        <%x>\n"
 			      "  jiffies              <%lx>\n"
 			      "  next_to_watch.status <%x>\n",
-				(unsigned long)((tx_ring - adapter->tx_ring) /
-					sizeof(struct e1000_tx_ring)),
+				(unsigned long)(tx_ring - adapter->tx_ring),
 				readl(hw->hw_addr + tx_ring->tdh),
 				readl(hw->hw_addr + tx_ring->tdt),
 				tx_ring->next_to_use,
